### ISSUE TITLE
Add `getByRiotId`, Add riot ID to summoner data

### ISSUE THREAD

**Issue by zbee** [meraki-analytics/cassiopeia#441]
[Riot Developer Relations update regarding Summoner Names/Riot IDs](https://www.riotgames.com/en/DevRel/summoner-names-to-riot-id).

> starting later this year ... All player-facing front-end fields and forms will require modification. Applications featuring the "Find your Summoner by Region + Name" functionality must adapt to locate summoners using Riot IDs, which are formed by combining the "game name" and "tag line".

> For third-party apps, displaying Riot IDs in place of summoner names within frontend fields is now necessary. 

[`/lol/summoner/v4/summoners/by-name/{summonerName}`](https://developer.riotgames.com/apis#summoner-v4/GET_getBySummonerName) is deprecated.

[`/lol/summoner/v4/summoners/by-puuid/{encryptedPUUID}`](https://developer.riotgames.com/apis#summoner-v4/GET_getByPUUID) is updated with Riot ID info.

[`riot/account/v1/accounts/by-riot-id/{gameName}/{tagLine}`](https://developer.riotgames.com/apis#account-v1/GET_getByRiotId) is new.

[`/riot/account/v1/accounts/by-puuid/{puuid}`](https://developer.riotgames.com/apis#account-v1/GET_getByPuuid) is new.

+a few other `/v4/summoners/...` tweaks

**Comment by zbee** [comment]
Working on adding the account endpoints [here](https://github.com/zbee/cassiopeia): from 060a19e onwards, but I cannot get the pipelines to provide the data yet.

**Comment by draymone** [comment]
Are your progressing on your branch ?

**Comment by AndreasLETANTER** [comment]
Hi, I'm trying to work on a project with Cassiopeia, but I couldn't retrieve summoner data based on their ID, like:

`get_summoner(name="name #tag", region="region")`

Is this branch made for supporting this ?

If yes, are you currently making progress on it?

**Comment by zbee** [comment]
That was the goal of the branch that I mention [above](https://github.com/meraki-analytics/cassiopeia/issues/441#issuecomment-1854657115) ([zbee/cassiopeia](https://github.com/zbee/cassiopeia/)) however I could not finish the support - there are too many places where the same code for an endpoint needs to be put, and I could not find them all, I think.

Currently that branch has all instances of endpoints added for an account endpoint with `get_account(puuid, gameName, Tagline)` to be added that you can find by ctrl+f'ing the project, but something is missing still and I don't know what or where. The function works, but it refuses to actually make the call to the API.

I am not currently making progress on this, no, nor do I plan to.
I switched to [Camille](https://github.com/MingweiSamuel/Camille) as it auto-generates the code for every endpoint, and as such of course supported the account endpoint and was the only major library at the time that did; [riot-watcher](https://github.com/pseudonym117/Riot-Watcher) still does not support account endpoints either as the main other python library. [Riven](https://github.com/MingweiSamuel/Riven) is the library I would choose if starting from scratch, being very similar in advantages to Camille but more active.

**Comment by stephen19K** [comment]
fellow Cass enjoyer here, feeling the pain of not being able to use `Summoner` object, therefore rendering much of the library unusable. It appears that Riot has fully removed the `summoners/by-name/` endpoint and related documentation.

Thanks for your efforts zbee! It's definitely a huge change required to migrate to Riot IDs. I would also love to contribute, but I will need to get more familiar with Cassiopeia's internals. I'm averse to switching as I'd love to use Python data viz libraries. I'll report back if I make any progress (eta: not soon tm).

**Comment by misiektoja** [comment]
Yes, the pain hit me as well as my code stopped working. I'm getting errors when trying to use Summoner object (+ earlier reported lack of Spectator-V4 API support). And there does not seem to be other Python lib supporting it as well (yet) :-(

**Comment by misiektoja** [comment]
Just FYI for all of you suffering from Cassiopeia still not supporting the recent RIOT API changes. I found out there is actually one Python lib that supports all the recent changes. It is called [Pulsefire](https://github.com/iann838/pulsefire). I just have rewritten my tool to use it, it took me few hours. It is maybe not that easy to use as Cassiopeia (requires some manual processing of data, no ready classes for some), but it works just fine and is actually very fast.

**Comment by jjmaldonis** [comment]
Hey folks, we will get this updated. Working on the code changes now.

**Comment by jjmaldonis** [comment]
Cass is now updated to support the Riot Account endpoints. The code is published on PyPi, version 5.1.0

The new syntax is:

```
account = Account(name="Polbelter", tagline="NA1", region="NA")
summoner = account.summoner
```

`summoner.name` no longer exists. Instead, use `account.name`, `account.tagline`, and/or `account.name_with_tagline`.

All the examples are updated, so check those out if you have questions.
